logger.go

```diff
@@ -1,40 +1,68 @@
 package server
 
 import (
+	"encoding/json"
 	"os"
 
 	"github.com/juju/errgo"
-
-	log "github.com/op/go-logging"
-	Stdlog "log"
+	"github.com/op/go-logging"
 )
 
 type LoggerOptions struct {
-	Name    string
+	ID      string
 	Level   string
 	NoColor bool
+	Meta    map[string]interface{}
+}
+
+type logFormat struct {
+	ID      string                 `json:"id"`
+	Time    string                 `json:"time"`
+	Level   string                 `json:"level"`
+	File    string                 `json:"file"`
+	Message string                 `json:"message"`
+	Meta    map[string]interface{} `json:"meta,omitempty"`
 }
 
 // NewSimpleLogger creates a new logger with a default backend logging to `os.Stderr`.
-func NewLogger(options LoggerOptions) *log.Logger {
-	// Create new logger.
-	logger := log.MustGetLogger(options.Name)
+func MustGetLogger(options LoggerOptions) *logging.Logger {
+	// Create new logger. The logger ID is used to identify the logger using the
+	// "id" field in the log format.
+	logger := logging.MustGetLogger(options.ID)
+
+	// See https://godoc.org/github.com/op/go-logging#NewStringFormatter for format verbs.
+	format := logFormat{
+		ID:      "%{module}",
+		Time:    "%{time}",
+		Level:   "%{level}",
+		File:    "%{longfile}",
+		Message: "%{message}",
+		Meta:    options.Meta,
+	}
 
 	// Configure logger.
-	log.SetFormatter(log.MustStringFormatter("[%{level}] %{message}"))
-	logBackend := log.NewLogBackend(os.Stderr, "", Stdlog.LstdFlags|Stdlog.Lshortfile)
-	logBackend.Color = !options.NoColor
-	log.SetBackend(logBackend)
+	rawFormat, err := json.Marshal(format)
+	if err != nil {
+		panic(errgo.Mask(err))
+	}
+
+	formatter := logging.MustStringFormatter(string(rawFormat))
+	backend := logging.NewLogBackend(os.Stderr, "", 0)
+	backend.Color = !options.NoColor
```

Comment:
The NoColor was added to prevent color output. Since the original "setting" was to have colors all the time, we would either change that, or keep it and introduce a flag that allows for being false by default.